enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -422,8 +422,10 @@ func (d *doubleReader) Read(p []byte) (n int, err error) {
 	d.bytesReadSrc += srcN
 
 	eg.Wait()
-	if dstErr != nil && dstErr != srcErr {
-		log.Warningf("Migration %v read err, src err: %s, dest err: %s, src read %d, dest read %d", d.r, srcErr, dstErr, d.bytesReadSrc, d.bytesReadDest)
+	// Don't log on EOF errors when reading chunks, because the readers from different caches
+	// could be reading at different rates. Only log on Close() if the total number of bytes read differs
+	if dstErr != nil && dstErr != srcErr && dstErr != io.EOF {
+		log.Warningf("Migration %v read err, src err: %s, dest err: %s", d.r, srcErr, dstErr)
```

Comment:
If the error could be nil, you can use %v instead of %s to print the nil value properly.

Reply:
Done